### PR TITLE
Image volumes

### PR DESCRIPTION
**Pull request by mrunalp**
We add support for mkdir and ignore to begin with.

### CODE REVIEW

**Comment by mrunalp** [comment]
I'll add some tests.

File: server/config.go [hunk 1]
```diff
@@ -145,6 +145,8 @@ type ImageConfig struct {
 	// InsecureRegistries is a list of registries that must be contacted w/o
 	// TLS verification.
 	InsecureRegistries []string `toml:"insecure_registries"`
+	// ImageVolumes controls how volumes specified in image config are handled
+	ImageVolumes string `toml:"image_volumes"`
```
Comment (rhatdan) [inline server/config.go]: Should we use a struct for this?  With constants?

File: server/container_create.go [hunk 2]
```diff
@@ -606,9 +606,17 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID string,
...
-			return nil, err1
+		switch s.config.ImageVolumes {
+		case "mkdir":
+			if err1 := os.MkdirAll(fp, 0644); err1 != nil {
+				return nil, err1
+			}
+		case "ignore":
+			logrus.Debugf("Ignoring volume %v", dest)
+		default:
+			logrus.Warnf("Unknown image volumes setting: %v", s.config.ImageVolumes)
```
Comment (rhatdan) [inline server/container_create.go]: This should probably be a failure, but this should be checked in the options parsings.

File: server/config.go (hunk 1 above)
Comment (runcom) [inline server/config.go]: +1 for constants, this is already a struct though `ImageConfig`

Comment (mrunalp) [inline server/config.go]: Sure, I'll add constants.

**Comment by mrunalp** [comment]
Updated.

**Comment by mrunalp** [comment]
@runcom This is ready now.

**Comment by runcom** [comment]
Tests looks fine as well, merging

**Pull request merged**